q2_moshpit/eggnog/_dbs.py

```diff
@@ -0,0 +1,89 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2022, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+import os
+from q2_types_genomics.reference_db import (
+    EggnogRefDirFmt, EggnogSequenceTaxaDirFmt
+)
+from .._utils import run_command, colorify
+
+
+def fetch_eggnog_db() -> EggnogRefDirFmt:
+    """
+    Downloads eggnog reference database using the
+    `download_eggnog_data.py` script from eggNOG. Here, this
+    script downloads 3 files amounting to 47Gb in total.
+    """
+
+    # Initialize output objects
+    eggnog_db = EggnogRefDirFmt()
+
+    # Define command.
+    # Meaning of flags:
+    # y: Answer yest to all prompts thrown by download_eggnog_data.py
+    # D: Do not download the Diamond database
+    # data_dir: location where to save downloads
+    cmd = [
+        "download_eggnog_data.py", "-y", "-D",
+        "--data_dir", str(eggnog_db.path)
+    ]
+    run_command(cmd)
+
+    # Return objects
+    return eggnog_db
+
+
+def fetch_eggnog_fasta() -> EggnogSequenceTaxaDirFmt:
```

Comment:
Maybe we can also rename this to `fetch-eggnog-proteins` to be more in-line with the name change I proposed in the types PR.